lib/es6Apply.js

```diff
@@ -0,0 +1,28 @@
+/* jshint esversion: 6 */
+(function(define) {
+'use strict';
+define(function() {
+	return function(func, thisObj, args) {
+		var result = null;
+
+		if(thisObj === func || (thisObj && thisObj.constructor === func)) {
+			/* jshint newcap: false */
+			result = new func(...(args||[]));
+
+			// detect broken old prototypes with missing constructor
+			if (result.constructor !== func) {
+				Object.defineProperty(result, 'constructor', {
```

Comment:
I don't want to modify the prototype from here, so I set the constructor property on object itself.